src/components/channel-effects/ChannelSlider.js

```diff
@@ -0,0 +1,75 @@
+import Slider from '@material-ui/core/Slider';
+import PubSub from 'pubsub-js';
+import { withStyles } from '@material-ui/core/styles';
+import React from 'react';
+import { COLORS_CHANGE } from '../../events';
+
+export default class ChannelSlider extends React.Component {
+  constructor(props) {
+    super(props);
+    const { channel } = props;
+    this.channel = channel;
+    this.state = {
+      sliderValue: [],
+      slider: null,
+      colorValue: [],
+    };
+    this.handleSliderChange = this.handleSliderChange.bind(this);
+  }
+
+  handleSliderChange(value) {
+    const channelValue = {};
+    channelValue[this.channel] = value;
+    this.setState({ sliderValue: value });
+    const { setSliderValue } = this.props;
+    setSliderValue(channelValue);
+  }
+
+  componentDidMount() {
+    this.handleSliderChange([0, 20000]);
+  }
+
+  componentWillMount() {
+    this.colorsChangeToken = PubSub.subscribe(COLORS_CHANGE, this.onColorsChange.bind(this));
+  }
+
+  componentWillUnmount() {
+    PubSub.unsubscribe(this.colorsChangeToken);
+  }
+
+  onColorsChange(msg, colorData) {
+    if (Object.keys(colorData)[0] === this.channel) {
```

Comment:
Is this like the case before, where we're expecting an object with a single key-value pair? I would prefer that it be an object with two keys, whose names explain what the key and value are.